pdns/bpf-filter.cc

```diff
@@ -354,6 +356,27 @@ BPFFilter::BPFFilter(std::unordered_map<std::string, MapConfiguration>& configs,
     throw std::runtime_error("Unsupported eBPF map format, the current internal implemenation only supports the legacy format");
   }
 
+  struct rlimit old_limit;
+  if (getrlimit(RLIMIT_MEMLOCK, &old_limit) != 0) {
+    throw std::runtime_error("Unable to get memory lock limit: " + stringerror());
+  }
+
+  const rlim_t minimal_limit_size = 64 * 124;
+  const rlim_t new_limit_size = 1024 * 1024;
+
+  /* Check if the current soft memlock limit is at least minimal_limit */
+  if (old_limit.rlim_cur < minimal_limit_size) {
```

Comment:
I think this is not very useful to raise the limit if the current value is below 64k, as most systems set `RLIMIT_MEMLOCK` to exactly 64k, and we know this is not enough for most usage (decent-sized maps). How about we try to raise that value to 1024k if it is below 1024k?

Reply:
That makes sense indeed